pkg/vpcmodel/nodesConnectivity.go

```diff
@@ -50,63 +53,90 @@ func (c *CloudConfig) GetVPCNetworkConnectivity(grouping bool) *VPCConnectivity
+		filterLayers := []string{NaclLayer, SecurityGroupLayer}
+		for _, layer := range filterLayers {
+			conns := c.getFiltersAllowedConnsBetweenNodesPerDirectionAndLayer(src, dst, isIngress, layer)
+			updatePerLayerRes(perLayerRes, layer, peerNode, conns)
 		}
+
 		if peerNode.IsInternal() {
 			// no need for router node, connectivity is from within VPC
 			// only check filtering resources
 			allowedConnsBetweenCapturedAndPeerNode := AllConns()
-			for _, filter := range c.FilterResources {
-				layerName := filter.Kind()
-				filteredConns := filter.AllowedConnectivity(src, dst, isIngress)
-				if _, ok := perLayerRes[layerName]; !ok {
-					perLayerRes[layerName] = map[Node]*common.ConnectionSet{}
-				}
-				perLayerRes[layerName][peerNode] = filteredConns
-				allowedConnsBetweenCapturedAndPeerNode = allowedConnsBetweenCapturedAndPeerNode.Intersection(filteredConns)
-				// do not break if empty, to enable computation for all layers
+			for _, resMap := range perLayerRes {
+				allowedConnsBetweenCapturedAndPeerNode = allowedConnsBetweenCapturedAndPeerNode.Intersection(resMap[peerNode])
 			}
 			allLayersRes[peerNode] = allowedConnsBetweenCapturedAndPeerNode
-			direction := "inbound"
-			if !isIngress {
-				direction = "outbound"
-			}
-			fmt.Printf("getAllowedConnsPerDirection: src: %s, dst %s, conn: %s, direction: %s\n", src.Cidr(), dst.Cidr(),
-				allowedConnsBetweenCapturedAndPeerNode.String(), direction)
-		} else { // else : external node -> consider attached routing resources
+		} else {
+			// else : external node -> consider attached routing resources
+
 			allowedConnsBetweenCapturedAndPeerNode := NoConns()
+			// node is associated with either a pgw or a fip
+			var appliedRouter RoutingResource
 			for _, router := range c.RoutingResources {
 				routerConnRes := router.AllowedConnectivity(src, dst)
 				if !routerConnRes.IsEmpty() { // connection is allowed through router resource
 					// TODO: consider adding connection attribute with details of routing through this router resource
 					allowedConnsBetweenCapturedAndPeerNode = routerConnRes
+					appliedRouter = router
+					updatePerLayerRes(perLayerRes, router.Kind(), peerNode, routerConnRes)
 				}
 			}
-			for _, filter := range c.FilterResources {
-				layerName := filter.Kind()
-				filteredConns := filter.AllowedConnectivity(src, dst, isIngress)
-				if _, ok := perLayerRes[layerName]; !ok {
-					perLayerRes[layerName] = map[Node]*common.ConnectionSet{}
-				}
-				perLayerRes[layerName][peerNode] = filteredConns
-				allowedConnsBetweenCapturedAndPeerNode = allowedConnsBetweenCapturedAndPeerNode.Intersection(filteredConns)
-				// do not break if empty, to enable computation for all layers
+			if appliedRouter == nil {
+				// without fip/pgw there is no external connectivity
+				allLayersRes[peerNode] = NoConns()
+				continue
+			}
+			// appliedFilters are either both nacl and sg (for pgw) or only sg (for fip)
```

Comment:
This comment might be IBM-specific. Consider rephrasing.

Reply:
done